### ISSUE TITLE
Error in if (k_init < k_min) stop("k_init cannot be less than k_min"): argument is of length zero

### ISSUE THREAD

**Issue by ryanhchung** [zdebruine/singlet#43]
Hi @zdebruine ,

Thanks fo solving the threads issue.

However, it arises the second issue, as i wrote.

Here I attach the code

Thank you!

```r
set.seed(123)
data <- NormalizeData(test) %>%
  singlet::RunNMF()

Error in if (k_init < k_min) stop("k_init cannot be less than k_min"): argument is of length zero
Traceback:

1. NormalizeData(test) %>% singlet::RunNMF()
2. singlet::RunNMF(.)
3. RunNMF.Seurat(.)
4. ard_nmf(A = A, k_init = k, n_replicates = reps, tol = tol, maxit = maxit, 
 .     verbose = verbose, L1 = L1, L2 = L2, threads = threads, test_density = test.set.density, 
 .     learning_rate = learning.rate, tol_overfit = tol.overfit, 
 .     trace_test_mse = trace.test.mse)

```

**Comment by zdebruine** [comment]
@ryanhchung Thanks!  I need a unit test for this.  I believe https://github.com/zdebruine/singlet/commit/e8b9effdbd06691b1bc0f1304db151f09b0c1f78 fixes the issue.  Can you confirm?

**Comment by ryanhchung** [comment]
HI @zdebruine, no, it isn't. Did you update the version? I tried to reinstall, but still the same version, and it seems like that is the reason. Thanks!

**Comment by ryanhchung** [comment]
Hi @zdebruine, unfortunately, no. Still same problem occurs. Thanks for your kind effort!

**Comment by zdebruine** [comment]
@ryanhchung Ok, I bumped the version now.  Can you try:

`
devtools::install_github("zdebruine/singlet")
library(singlet)
pbmc3k <- get_pbmc3k_data()
pbmc3k <- RunNMF(pbmc3k)
`

Does that work for you?  (it does for me).  If it does, can you check your issue as well?

**Comment by ryanhchung** [comment]
Seems like it is working! Thank you so much!

Just to clarify, if I have pre-processed Seurat object, then I can skip the NormalizdData step, right?

Thank you!

**Comment by zdebruine** [comment]
Yes, if you have normalized data in the `@data` slot, you can skip the NormalizeData step.
